internal/idutils/external_id.go

```diff
@@ -50,3 +50,23 @@ func GenerateExternalID(c coordinate.Coordinate) (string, error) {
 }
 
 type ExternalIDGenerator func(coordinate.Coordinate) (string, error)
+
+// GenerateExternalIDForDocument generates an external ID for a document configuration. It is under 50 characters long and uses at most only "a-z", "A-Z", "0-9" and "-".
+func GenerateExternalIDForDocument(c coordinate.Coordinate) (string, error) {
```

Comment:
Could you rename `GenerateExternalID` so that name indicate it's just for Settings.